beacon-chain/blockchain/forkchoice/process_attestation.go

```diff
@@ -166,12 +166,29 @@ func (s *Store) verifyAttestation(ctx context.Context, baseState *pb.BeaconState
 	if err != nil {
 		return nil, errors.Wrap(err, "could not convert attestation to indexed attestation")
 	}
+	// check slashing
+	go s.checkForSlashing(ctx, indexedAtt)
```

Comment:
I would avoid doing this. We should feed the attestation from sync service to slasher service and run it there. It's important to keep the state transition pure and well contained. The cost of converting attestation to indexed attestation is almost non-negligible so it really makes no sense to plug slasher here